### DISCUSSION TITLE
Sizing for tutorials (newbie)

### DISCUSSION THREAD

**Discussion by eostermueller** [keptn/keptn#3692] [Q&A] [answered]
Hi, thanks for sharing keptn with us.....using 0.8.1 alpha with kubernetes 1.20.

The k8s pods are up and running but the Istio pods are all stuck on pending after I ran this:
```istioctl install --set profile=demo -y```

These two links suggest I'm lacking hardware, but  I've got a single node with 8cpu/32g RedHat 8 machine -- thought this was Keptn's recommendation.  Here are two posts that suggest my node is undersized.

https://discuss.istio.io/t/istio-pilot-component-pod-is-in-pending-state/299
https://github.com/istio/istio/issues/9083

Perhaps the "cpu" and "memory" requirements detailed in the profile are asking for more than the 32g/8cpu I've got?

https://github.com/istio/istio/blob/master/manifests/profiles/demo.yaml
I've looked for a reference, but I don't understand the units of measure in demo.yaml, like "10m" CPU and Memory: 100mi and 40mi.
Below shows the current status and thanks in advance for any help.

--Erik

```
	[ec2-user@ip-172-31-22-45 tutorial_prometheus]$ kubectl -n kube-system get deployments
	NAME      READY   UP-TO-DATE   AVAILABLE   AGE
	coredns   2/2     2            2           167m

	[ec2-user@ip-172-31-22-45 tutorial_prometheus]$ kubectl -n kube-system get pods
	
	NAME                                                                 READY   STATUS    RESTARTS   AGE
	coredns-74ff55c5b-bdvlg                                              1/1     Running   0          169m
	coredns-74ff55c5b-jl4tk                                              1/1     Running   0          169m
	etcd-ip-172-31-22-45.us-east-2.compute.internal                      1/1     Running   0          169m
	kube-apiserver-ip-172-31-22-45.us-east-2.compute.internal            1/1     Running   0          169m
	kube-controller-manager-ip-172-31-22-45.us-east-2.compute.internal   1/1     Running   0          169m
	kube-proxy-rjrtf                                                     1/1     Running   0          169m
	kube-scheduler-ip-172-31-22-45.us-east-2.compute.internal            1/1     Running   0          169m
	weave-net-n7dtt                                                      2/2     Running   1          155m
	[ec2-user@ip-172-31-22-45 tutorial_prometheus]$

	[ec2-user@ip-172-31-22-45 tutorial_prometheus]$ k get pods -n istio-system
	NAME                                   READY   STATUS    RESTARTS   AGE
	istio-egressgateway-bd6d77495-w72wc    0/1     Pending   0          56m
	istio-ingressgateway-d46c4ff9b-hp57b   0/1     Pending   0          56m
	istiod-79ddd9dbf5-l4m22                0/1     Pending   0          61m


	[ec2-user@ip-172-31-22-45 tutorial_prometheus]$ k get events -n istio-system --sort-by=".metadata.managedFields[0].time"
	LAST SEEN   TYPE      REASON              OBJECT                                      MESSAGE
	16m         Warning   FailedScheduling    pod/istiod-79ddd9dbf5-l4m22                 0/1 nodes are available: 1 node(s) had taint {node-role.kubernetes.io/master: }, that the pod didn't tolerate.
	58m         Warning   FailedScheduling    pod/istio-egressgateway-bd6d77495-w72wc     0/1 nodes are available: 1 node(s) had taint {node-role.kubernetes.io/master: }, that the pod didn't tolerate.
	58m         Normal    SuccessfulCreate    replicaset/istio-ingressgateway-d46c4ff9b   Created pod: istio-ingressgateway-d46c4ff9b-hp57b
	58m         Normal    SuccessfulCreate    replicaset/istio-egressgateway-bd6d77495    Created pod: istio-egressgateway-bd6d77495-w72wc
	58m         Normal    ScalingReplicaSet   deployment/istio-egressgateway              Scaled up replica set istio-egressgateway-bd6d77495 to 1
	58m         Warning   FailedScheduling    pod/istio-ingressgateway-d46c4ff9b-hp57b    0/1 nodes are available: 1 node(s) had taint {node-role.kubernetes.io/master: }, that the pod didn't tolerate.
	58m         Normal    ScalingReplicaSet   deployment/istio-ingressgateway             Scaled up replica set istio-ingressgateway-d46c4ff9b to 1
	58m         Normal    NoPods              poddisruptionbudget/istio-ingressgateway    No matching pods found
	58m         Normal    NoPods              poddisruptionbudget/istio-egressgateway     No matching pods found
	16m         Warning   FailedScheduling    pod/istio-ingressgateway-d46c4ff9b-hp57b    0/1 nodes are available: 1 node(s) had taint {node-role.kubernetes.io/master: }, that the pod didn't tolerate.
	16m         Warning   FailedScheduling    pod/istio-egressgateway-bd6d77495-w72wc     0/1 nodes are available: 1 node(s) had taint {node-role.kubernetes.io/master: }, that the pod didn't tolerate.

```

**Comment by eostermueller** [comment, marked as answer]
Replying to self here, got this fixed.  My friend Raj helped out here.
Had two problems.  Since this is a poc, trying to run everything on 1 node.
Had to somehow get "Unschedulable:true" to false, and had to remove a 'taint'.
See below for details.

--Erik

	[ec2-user@myNode tutorial_prometheus]$ k describe nodes | more
	Name:               myNode.us-east-2.compute.internal
	Roles:              control-plane,master
	Labels:             beta.kubernetes.io/arch=amd64
						beta.kubernetes.io/os=linux
						kubernetes.io/arch=amd64
						kubernetes.io/hostname=myNode.us-east-2.compute.internal
						kubernetes.io/os=linux
						node-role.kubernetes.io/control-plane=
						node-role.kubernetes.io/master=
	Annotations:        kubeadm.alpha.kubernetes.io/cri-socket: /var/run/dockershim.sock
						node.alpha.kubernetes.io/ttl: 0
						volumes.kubernetes.io/controller-managed-attach-detach: true
	CreationTimestamp:  Mon, 05 Apr 2021 00:23:26 +0000
	Taints:             node-role.kubernetes.io/master:NoSchedule
						node.kubernetes.io/unschedulable:NoSchedule
	Unschedulable:      true



FIX #1:	

	[ec2-user@myNode tutorial_prometheus]$ k uncordon myNode.us-east-2.compute.internal
	node/myNode.us-east-2.compute.internal uncordoned


VALIDATE:  The following shows the "Uncheduable" is not set to false.
	Taints:             node-role.kubernetes.io/master:NoSchedule
	Unschedulable:      false


FIX #2:	 

k taint node myNode.us-east-2.compute.internal node-role.kubernetes.io/master:NoSchedule-
node/myNode.us-east-2.compute.internal untainted

**Reply by johannes-b** [reply]
Hi Erik, 
many thanks for sharing the fix to your problem. This helps the community!
